ons-spark/ancillary-topics/pandas_udfs_rescue.ipynb

```diff
@@ -0,0 +1,902 @@
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>incident_number</th>\n",
+       "      <th>minute_cost(incident_duration, incident_cost)</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>72214141</td>\n",
+       "      <td>9.833333</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>165500101</td>\n",
+       "      <td>4.333333</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>129048101</td>\n",
+       "      <td>4.333333</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>129261101</td>\n",
+       "      <td>8.666667</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>2872091</td>\n",
+       "      <td>4.250000</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>5</th>\n",
+       "      <td>40866091</td>\n",
+       "      <td>8.500000</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>6</th>\n",
+       "      <td>76285091</td>\n",
+       "      <td>4.333333</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>7</th>\n",
+       "      <td>87402091</td>\n",
+       "      <td>4.333333</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>8</th>\n",
+       "      <td>89957091</td>\n",
+       "      <td>4.333333</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>9</th>\n",
+       "      <td>102278091</td>\n",
+       "      <td>8.666667</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "  incident_number  minute_cost(incident_duration, incident_cost)\n",
+       "0        72214141                                       9.833333\n",
+       "1       165500101                                       4.333333\n",
+       "2       129048101                                       4.333333\n",
+       "3       129261101                                       8.666667\n",
+       "4         2872091                                       4.250000\n",
+       "5        40866091                                       8.500000\n",
+       "6        76285091                                       4.333333\n",
+       "7        87402091                                       4.333333\n",
+       "8        89957091                                       4.333333\n",
+       "9       102278091                                       8.666667"
+      ]
+     },
+     "execution_count": 17,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# A simple Pandas scalar UDF to find out how much each minute cost per incident.\n",
```

Comment:
I think it would be best to use examples of things we can't do very easily in PySpark. I can't think of anything off the top of my head, but let's have a think and come back to this. Could do with some input from Sam/Rowan/David/me.